ambry-rest/src/main/java/com/github/ambry/rest/RestServer.java

```diff
@@ -206,9 +206,17 @@ public RestServer(VerifiableProperties verifiableProperties, ClusterMap clusterM
       routerSslFactory = sslFactory;
     }
 
-    RouterFactory routerFactory =
-        Utils.getObj(restServerConfig.restServerRouterFactory, verifiableProperties, clusterMap, notificationSystem,
-            routerSslFactory, accountService);
+    RouterFactory routerFactory;
+    if (restServerConfig.restServerEnableCompositeRouter) {
+      RouterFactory ambryRouterFactory =
+          Utils.getObj(restServerConfig.restServerRouterFactory, verifiableProperties, clusterMap,
+              notificationSystem, routerSslFactory, accountService);
+      Router ambryRouter = ambryRouterFactory.getRouter();
+      routerFactory = Utils.getObj(restServerConfig.restServerCompositeRouterFactory, ambryRouter);
```

Comment:
This is good. I was just looking at [CompositeAccountServiceFactory](https://github.com/linkedin/ambry/blob/master/ambry-account/src/main/java/com/github/ambry/account/CompositeAccountServiceFactory.java#L46) example. Similar to it, we could also have `restServerPrimaryRouterFactory` and `restServerSecondaryRouterFactory`. Then, we wouldn't need `restServerEnableCompositeRouter` config. We could directly point `restServerRouterFactory` to `CompositeRouterFactory` in the configs to enable composite router.